lib/Marlin/Marlin/src/gcode/lcd/M73_PE.cpp

```diff
@@ -102,6 +103,22 @@ void M73_PE_no_parser(std::optional<uint8_t> P, std::optional<uint32_t> R, std::
     if (T) {
         oProgressData.oTime2Pause.mSetValue(*T, nTimeNow); // [min] -> [s]
     }
+
+    if (!P && !R && !T) {
+        SERIAL_ECHO_START();
+        SERIAL_ECHOLNPAIR(" M73 Progress: ", marlin_vars()->sd_percent_done, "%;");
+        uint32_t timetoend = marlin_vars_t().time_to_end;
+        const char minutes_str[3] = "m;";
+        if (timetoend != marlin_server::TIME_TO_END_INVALID) {
+            SERIAL_ECHOPAIR(" Time left: ", timetoend / 60, minutes_str);
+            SERIAL_EOL();
+        }
+        uint32_t timetopause = oProgressData.oTime2Pause.mGetValue();
+        if (timetopause != marlin_server::TIME_TO_END_INVALID) {
+            SERIAL_ECHOPAIR(" Change: ", ((timetopause * 100) / marlin_vars()->print_speed) / 60, minutes_str);
```

Comment:
This may divide by zero, please guard against that.

Where does the 100 come from?

I wouldn't factor out minutes_str like that, compiler should be perfectly capable of doing that and it doesn't add to readability.

`timetoend` and `timetopause` can be declared `const`. I would also prefer `time_to_end` and `time_to_pause`

Please, add comment similar to https://github.com/prusa3d/Prusa-Firmware-Buddy/pull/3770 to let future maintainers know that changing this may break external tools depending on parsing messages from serial line.

Reply:
The 100 is coming from this:  https://github.com/prusa3d/Prusa-Firmware-Buddy/blob/5fc26ee526a57b54ee5583100908522c2b3bb4b1/src/common/marlin_server.cpp#L2514-L2515

Added the divide/zero guard, renamed variables and declared const, factored out minutes_str and also added the comment.